app/forms/date_form.py

```diff
@@ -87,6 +88,44 @@ class MonthYearDateForm(Form):
     return MonthYearDateForm
 
 
+def get_year_form(answer, answer_store, metadata, error_messages, label, guidance):
+    """
+    Returns a month year form metaclass with appropriate validators. Used in both date and
+    date range form creation.
+
+    :param answer: The answer on which to base this form
+    :param error_messages: The messages to use upon this form during validation
+    :return: The generated MonthYearDateForm metaclass
+    """
+
+    class YearDateForm(Form):
+        year = None
+
+    validate_with = [OptionalForm()]
+
+    if answer['mandatory'] is True:
+        validate_with = validate_mandatory_date(error_messages, answer)
+
+    if 'validation' in answer and 'messages' in answer['validation'] \
+            and 'INVALID_DATE' in answer['validation']['messages']:
```

Comment:
This code is used in each `get_form` list, probably worth putting in a method at bottom and call, `is_bespoke_message` or something